grpc-native/src/main/java/org/ballerinalang/net/grpc/listener/ServerCallHandler.java

```diff
@@ -182,9 +192,10 @@ void onMessageInvoke(ServiceResource resource, Message request, StreamObserver r
         List<Type> signatureParams = resource.getParamTypes();
         Object[] paramValues;
         int i = 0;
-        if (signatureParams.size() >= 1 && CALLER_TYPE.equals(signatureParams.get(0).getName())) {
+        if ((signatureParams.size() >= 1) && (signatureParams.get(0).getTag() == TypeTags.OBJECT_TYPE_TAG) &&
```

Comment:
This is the logic I came up with to differentiate 
1. remote functions that return the data and
2. remote functions that use callers

@daneshk please check the validity.

Reply:
This is ok. The `Caller` object always comes as the first param. We can validate this in the compiler plugin.